web/src/compose_ui.js

```diff
@@ -375,6 +385,122 @@ export function format_text($textarea, type, inserted_content) {
         selected_text.slice(0, bold_syntax.length) === bold_syntax &&
         selected_text.slice(-bold_syntax.length) === bold_syntax;
 
+    const section_off_selected_lines = () => {
+        // Divide all lines of text (separated by `\n`) into those entirely or
+        // partially selected, and those before and after these selected lines.
+        const before = text.slice(0, range.start);
+        const after = text.slice(range.end);
+        let separating_new_line_before = false;
+        let closest_new_line_beginning_before_index;
+        if (before.includes("\n")) {
+            separating_new_line_before = true;
+            closest_new_line_beginning_before_index = before.lastIndexOf("\n");
+        } else {
+            separating_new_line_before = false;
+            // The beginning of the entire text acts as a new line.
+            closest_new_line_beginning_before_index = -1;
+        }
+        let separating_new_line_after = false;
+        let closest_new_line_char_after_index;
+        if (after.includes("\n")) {
+            separating_new_line_after = true;
+            closest_new_line_char_after_index =
+                after.indexOf("\n") + before.length + selected_text.length;
+        } else {
+            separating_new_line_after = false;
+            // The end of the entire text acts as a new line.
+            closest_new_line_char_after_index = text.length;
+        }
+        // selected_lines neither includes the `\n` character that marks it's
+        // beginning (which exists if there are before_lines) nor the one
+        // that marks it's end (which exists if there are after_lines).
```

Comment:
```suggestion
        // that marks its end (which exists if there are after_lines).
```

Reply:
Tiniest of tiny nits: Possessive _its_ doesn't take an apostrophe

Reply:
Thanks for catching this! This is one of those things where I know it's wrong, but by force of habit, I make the mistake anyway :see_no_evil: